label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleValidatorImpl.java

```diff
@@ -264,6 +265,9 @@ public void validateLabelExpression(Action.DynamicLabel dynamicLabel) {
   private void validateTokenExtractionRule(
       Action.DynamicLabel.TokenExtractionRule tokenExtractionRule) {
     validateNonDefaultPresenceOrThrow(tokenExtractionRule, tokenExtractionRule.KEY_FIELD_NUMBER);
+    if (tokenExtractionRule.hasRegexCapture()) {
+      RegexValidator.validateCaptureGroupCount(tokenExtractionRule.getRegexCapture(), 1);
```

Comment:
Why are we restricting only to one capture group?

Reply:
Since we need a single value, we can use value only from one group

Reply:
You'd create create two rules with one capture group each.

Reply:
But that would lead to two different labels, right? If we want both of of them in a single label?

Reply:
No - a dynamic label defines tokens and an expression that indicates how to combine them - this PR is adding a new way to extract a token:
```proto
    message DynamicLabel {
      string label_expression = 1; // expression to combine tokens to generate label, example "${deployment}_${version}"
      repeated TokenExtractionRule token_extraction_rules = 2;
      message TokenExtractionRule {
        string key = 1;
        optional string json_path = 2;
        optional string alias = 3;
        optional string regex_capture = 4;
      }
    }
```

Reply:
Oh, okay